test/util/framework/hcp_helper.go

```diff
@@ -16,17 +16,154 @@ package framework
+	}
+
+	return nil
+}
+
+func VerifyImageRegistryDisabled() HostedClusterVerifier {
+	return verifyImageRegistryDisabled{}
+}
+
+type verifyBasicAccessImpl struct{}
+
+func (v verifyBasicAccessImpl) Name() string {
+	return "VerifyBasicAccess"
+}
+
+func (v verifyBasicAccessImpl) Verify(ctx context.Context, adminRESTConfig *rest.Config) error {
+	kubeClient, err := kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create kubernetes client: %w", err)
+	}
+
+	_, err = kubeClient.CoreV1().Services("default").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list services: %w", err)
+	}
+
+	return nil
+}
+
+func verifyBasicAccess() HostedClusterVerifier {
+	return verifyBasicAccessImpl{}
+}
+
+var standardVerifiers = []HostedClusterVerifier{
+	verifyBasicAccess(),
+}
+
+func VerifyHCPCluster(ctx context.Context, adminRESTConfig *rest.Config, additionalVerifiers ...HostedClusterVerifier) error {
+	allVerifiers := append(standardVerifiers, additionalVerifiers...)
+
+	// if these start taking a long time, run in parallel
+	errs := []error{}
+	for _, verifier := range allVerifiers {
+		err := verifier.Verify(ctx, adminRESTConfig)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("%v failed: %w", verifier.Name(), err))
+		}
+	}
+
+	return errors.Join(errs...)
+}
+
+func GetAdminRESTConfigForHCPCluster(
+	ctx context.Context,
+	hcpClient *hcpapi20240610.HcpOpenShiftClustersClient,
+	resourceGroupName string,
+	hcpClusterName string,
+	timeout time.Duration,
+) (*rest.Config, error) {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	adminCredentialRequestPoller, err := hcpClient.BeginRequestAdminCredential(
+		ctx,
+		resourceGroupName,
+		hcpClusterName,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to start credential request: %w", err)
+	}
+
+	operationResult, err := adminCredentialRequestPoller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{
+		Frequency: StandardPollInterval,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed waiting for hcpCluster=%q in resourcegroup=%q to finish getting creds: %w", hcpClusterName, resourceGroupName, err)
+	}
+
+	switch m := any(operationResult).(type) {
+	case hcpapi20240610.HcpOpenShiftClustersClientRequestAdminCredentialResponse:
+		return readStaticRESTConfig(m.Kubeconfig)
+	default:
+		return nil, fmt.Errorf("unknown type %T", m)
+	}
+}
+
+func readStaticRESTConfig(kubeconfigContent *string) (*rest.Config, error) {
+	ret, err := clientcmd.BuildConfigFromKubeconfigGetter("", func() (*clientcmdapi.Config, error) {
+		if kubeconfigContent == nil {
+			return nil, fmt.Errorf("kubeconfig content is nil")
+		}
+		return clientcmd.Load([]byte(*kubeconfigContent))
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// we are doing this because there's a serious bug.  I haven't got an ETA on a fix, but if we fail to correct it, we definitely need to know.
```

Comment:
Maybe just leave a TODO here and link this https://issues.redhat.com/browse/XCMSTRAT-950 instead of the slack thread?